web/client/components/theme/ContextTheme.jsx

```diff
@@ -32,8 +36,20 @@ function ContextTheme({
     version
 }) {
     const validatedVersion = validateVersion(version) ? trim(version) : '';
+    let newVars = "";
+    try {
+        less.render(themeVars + ".get-root-css-variables(@ms-theme-vars);", {
+            modifyVars: theme?.[0]?.colors || {}
```

Comment:
This update defines the theme property as an object so we could change the selector to not cast theme as an array.
Then we should also use the `variables` property instead of `colors` inside the saved theme and use a default or init + Variables to save the base variables